techcsispit/House-Price-Prediction
# Add Prediction Page and Set Up React Router

**Description**
This PR introduces the following changes to enhance the application:

**Added Prediction Page:** A new prediction page has been created to allow users to submit data and receive predictions.

**Implemented React Router:** The application now uses React Router for navigation, providing a seamless user experience between the dashboard and the prediction page.

**Added requirements.txt:** A requirements.txt file has been added to the backend to manage Python package dependencies that were previously missing.

**How to Test**
Frontend:
Make sure backend is running 
Run the frontend application using npm run dev.
Navigate to the dashboard and click on the button get started to go to the prediction page.
Verify that the prediction page loads correctly and that the form functions as expected.


Working Demo 
![image](https://github.com/user-attachments/assets/8f4b1b1f-0240-4344-8992-49c7ee60cf55)

**Benefits**
The new prediction page provides a dedicated space for users to input their data, enhancing usability.
Implementing form validation ensures data integrity and reduces the likelihood of errors during submission.
Default values help guide users, making the form easier to complete and improving the overall user experience.
